kong/plugins/ai-proxy/schema.lua

```diff
@@ -0,0 +1,125 @@
+local typedefs = require "kong.db.schema.typedefs"
+
+local auth_schema = {
+  type = "record",
+  required = false,
+  fields = {
+    { header_name = { type = "string", description = "If AI model requires authentication via Authorization or API key header, specify its name here.", required = false, referenceable = true } },
+    { header_value = { type = "string", description = "Specify the full auth header value for 'header_name', for example 'Bearer key' or just 'key'.", required = false, referenceable = true } },
+    { param_name = { type = "string", description = "If AI model requires authentication via query parameter, specify its name here.", required = false, referenceable = true } },
+    { param_value = { type = "string", description = "Specify the full parameter value for 'param_name'.", required = false, referenceable = true } },
+    { param_location = { type = "string", description = "Specify whether the 'param_name' and 'param_value' options go in a query string, or the POST form/JSON body.", required = false, one_of = { "query", "body" } } },
+  }
+}
+
+local model_options_schema = {
+  description = "Key/value settings for the model",
+  type = "record",
+  required = false,
+  fields = {
+    { max_tokens = { type = "integer", description = "Defines the max_tokens, if using chat or completion models.", required = false, default = 256 } },
+    { temperature = { type = "number", description = "Defines the matching temperature, if using chat or completion models.", required = false, between = { 0.0, 5.0 }, default = 1.0 } },
+    { top_p = { type = "number", description = "Defines the top-p probability mass, if supported.", required = false, between = { 0, 1 }, default = 1.0 } },
+    { top_k = { type = "integer", description = "Defines the top-k most likely tokens, if supported.", required = false,between = { 0, 500 }, default = 0 } },
+    { anthropic_version = { type = "string", description = "Defines the schema/API version, if using Anthropic provider.", required = false } },
+    { azure_instance = { type = "string", description = "Instance name for Azure OpenAI hosted models.", required = false } },
+    { llama2_format = { type = "string", description = "If using llama2 provider, select the upstream message format.", required = false, one_of = { "raw", "openai"} } },
+    { upstream_url = typedefs.url { description = "Manually specify or override the full URL to the AI operation endpoints, when calling (self-)hosted models, or for running via a private endpoint.", required = false } },
+  }
+}
+
+local model_schema = {
+  type = "record",
+  required = true,
+  fields = {
+    { provider = { type = "string", description = "AI provider request format - Kong translates requests to and from the specified backend compatible formats.", required = true, one_of = { "openai", "azure", "anthropic", "cohere", "mistral", "llama2", "preserve" }, default = "openai" } },
```

Comment:
Please go through all the Lua tables in this file and reformat to reduce line lengths to 80-ish characters. No strict prescriptions on exactly _how_ everything is formatted, just use your best judgment on readability (and readability of future diffs when the code is updated). Here are some notes:

1. You don't necessarily need to push _every_ element to its own line to shorten the line lengths. This is probably fine:

```lua
local my_table = { "one", "two", "three", "four", "five",
                   "six", "seven", "eight" }
```

2. Extending a string literal to > 80 chars is fine, but consider using string concatenation if it's _super_ long:

```lua
local my_long_string = "This is a super duper really extra super amazingly ultra long "
                    .. "string. Thankfully it's a bit more readable in the text editor "
                    .. "because it's broken up into multiple lines."

Reply:
Done